docs/source/conventions/cuda_conventions.rst

```diff
@@ -1,44 +1,67 @@
 
 1D Grids
 ^^^^^^^^
 
-Kernel functions that depend on the number of input gaussians are organized into one dimensional CUDA grids consisting of 
+Kernel functions that depend on the number of input Gaussians are organized into a CUDA grid consisting of
 
 .. code-block::
 
-    const int gridDim = (num_gaussian_operations + NUM_THREADS - 1) / NUM_THREADS;
+    const int gridDim = (num_gaussian_operations + N_THREADS - 1) / N_THREADS;
     
-1-D blocks containing ``const int blockDim = NUM_THREADS;`` threads each. An example of a kernel function that depends on the number of gaussians (here denoted by the ``num_gaussian_operations`` variable)
-is the main `project_gaussians` kernel that projects an arbitrary number of 3D gaussians into 2D.
+1D blocks containing ``const int blockDim = N_THREADS;`` threads each. An example of a kernel function that depends on the number of Gaussians (here denoted by the ``num_gaussian_operations`` variable)
+is the main `project_gaussians` kernel that projects an arbitrary number of 3D Gaussians into 2D.
 
 
 2D Grids
 ^^^^^^^^
 
-Kernel functions that depend on the rendered output image size are organized into two dimensional CUDA grids. The shape of the grid is determined by 
+Kernel functions that depend on the rendered output image size are organized into 2D CUDA grids. The shape of the grid is determined by 
 
-.. code-block:: 
+.. code-block::
     
-    const dim3 gridDim = {(img_width + NUM_THREADS_X - 1) / NUM_THREADS_X, (img_height + NUM_THREADS_Y - 1) / NUM_THREADS_Y, 1};
+    const dim3 gridDim = {(img_width + block_width - 1) / block_width, (img_height + block_width - 1) / block_width, 1};
 
-where each individual block within the grid contains a layout of threads defined by ``const dim3 blockDim = {NUM_THREADS_X, NUM_THREADS_Y, 1};``.
-An example of a kernel function that requires two dimensional grids is the main :func:`gsplat.rasterize_gaussians` kernel that renders an arbitrary number of pixels in an output image.
+where each individual block within the grid contains a layout of threads defined by ``const dim3 blockDim = {block_width, block_width, 1};``. Note that ``block_width * block_width * 1 <= N_THREADS``.
+An example of a kernel function that requires 2D grids is the main `rasterize_Gaussians` kernel that renders an arbitrary number of pixels in an output image.
 
 
 Config Constants
 ----------------
 
-We fix the total number of threads within a block to 256. This means that for 1D blocks the block dimension is given by ``const int blockDim = 256;`` and for 2D blocks ``const dim3 blockDim = {16, 16, 1};``.
+We fix the total number of threads within a block to ``N_THREADS = 256``. This is compatible with most CUDA devices. For 1D blocks the block dimension is then given by ``const int blockDim = 256;`` and for 2D blocks we set ``block_width = 16`` by default such that the blockDim is ``const dim3 blockDim = {16, 16, 1};``.
```

Comment:
This is technically not true anymore, the user can set block dim to be anywhere from 2x2 to 16x16 from the python API (enables rasterizing high dimension features)

Reply:
@kerrj, that is true. I have some doubts about it, mainly because I don't understand shared memory well enough. Do you have any answers to [this](https://github.com/nerfstudio-project/gsplat/pull/129#issuecomment-1987257215)? Just trying to understand it a bit better :) thanks

Reply:
@maturk Do you want to update this PR accordingly and merge? I think improving docs is a nice thing to do, we shouldn't discard this effort.

Reply:
@jb-ye, thanks! I will try to update the docs taking into account all the latest changes once I get the time again :)